### DISCUSSION TITLE
LICENSE for the demo code

### DISCUSSION THREAD

**Discussion by Tobi-De** [Instawork/hyperview#804] [Q&A] [answered]
Hello, first of all, thank you to everyone involved in this remarkable project. I'm a big enthusiast of htmx, which led me to discover hyperview.

I'm wondering if there are any plans to introduce a `create-hyperview-app` project starter? Currently, the primary method I've found to set up a hyperview project involves cloning the hyperview repository, extracting the demo app, and renaming the folder (e.g., to 'mobile_app'), as I usually integrate it into another project, such as a Django project. I'm interested in automating this process. Some time ago, I created a repository at https://github.com/Tobi-De/hyperview-starter, and I'm revisiting it now to update and automate how I synchronize it with the main repository (I'm considering using GitHub actions and some python scripts).

Reflecting on the original demo code for my purposes, I realize that I may not be fully adhering to the license terms. That's why I'm here. There is licensing text at the beginning of every file in the demo `src` folder. If I need to comply with it, does that mean I would need to include the LICENSE file in every project I create based on the demo code? If so, then the instructions in the demo README probably need to be updated to include a line about copying the LICENSE file as well; otherwise, the licensing text at the top of the files in the `src` folder wouldn't make sense.

If my understanding is correct, then I need to include the LICENSE in my starter project and all the code generated by it. Including it in my starter project is understandable, but for all the code generated by it, that would be unusual for most people. It would be akin to generating a React app with `create-react-app` or a Django project with `django-admin` and having a LICENSE file in my project.

So, I suppose my question is, how can I manage this while still respecting the work and the LICENSE of hyperview?

**Comment by Tobi-De** [comment]
Ok, I should have looked that up before, sorry about that.
My first question is already answered by https://github.com/Instawork/hyperview/issues/451
Perhaps I should work on this issue directly?

**Reply by flochtililoch** [reply, marked as answer]
hi @Tobi-De - regarding the license headers present in every files, ignore them. I have a task I'm planning to address soon, that will take care of removing them all. Only consider the LICENSE file at the root of the project, which is the standard MIT License.
The snippet I gave you on #451 doesn't need licensing from what I understand, so it shouldn't be a problem for your projects that use it. But obviously as the snippet relies on the Hyperview dependency, if that dependency ends up being embedded when distributed (i.e. not just referred to in a package.json), then the license would need to be kept along with the rest of the Hyperview files.
I'm not an expert on the topic, but you can likely find documentation around how to properly comply to the MIT License.

**Reply by flochtililoch** [reply]
@Tobi-De FYI: #813
